### PR TITLE
Remove mocks and improve results page test coverage 

### PR DESCRIPTION
**Pull request by MartinBenediktBusch**
### Overview 

- removed all mocks from `resultPage.spec.ts`
- refactored `resultPage.ts` to allow easy testing. I.e. separated query execution from api calls. 
- implemented basic tests that test the query results 
- fixed a bug identified by the tests related to the separation of strings. I.e. now it is allowd that affiliation names contain commas.

### CODE REVIEW

File: src/services/statistics.spec.ts [hunk 1]
```diff
@@ -1,50 +1,98 @@
...
 import { cleanup, seed } from '../utils/db/seed';
-import { Request, Response } from 'express';
+import { z } from 'zod';
+import { executeResultQueries } from './statistics';
+import { eq } from 'drizzle-orm';
 
 const DB_CONNECTION_URL = 'postgresql://postgres:secretpassword@localhost:5432';
 
-describe('service: statistics', () => {
+describe('getResultStatistics endpoint', () => {
```
Comment (diegoalzate) [inline src/services/statistics.spec.ts:14]: rename back to service: statistics

File: src/services/statistics.spec.ts [hunk 2]
```diff
@@ -1,50 +1,98 @@
...
+    };
+    otherUserTestData = {
+      numOfVotes: 2,
+      optionId: questionOption?.id ?? '',
+      questionId: forumQuestion?.id ?? '',
+      userId: otherUser?.id ?? '',
+    };
+
+    // Add additional data to the Db
+    await dbPool.update(db.cycles).set({ status: 'OPEN' }).where(eq(db.cycles.id, cycle!.id));
```
Comment (diegoalzate) [inline src/services/statistics.spec.ts:52]: i think the seed now auto makes the cycle open

**Review by diegoalzate — APPROVED** [review]
looks good, just left some small changes

File: src/services/statistics.spec.ts (hunk 1 above)
Comment (MartinBenediktBusch) [reply src/services/statistics.spec.ts]: implemented

File: src/services/statistics.spec.ts (hunk 2 above)
Comment (MartinBenediktBusch) [reply src/services/statistics.spec.ts]: removed

**Pull request merged**